src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/CSVConverter.java

```diff
@@ -0,0 +1,252 @@
+    private String ncksPath = null;
+    private String ncpdqPath = null;
+    private final String mimeType = "text/csv";
+    private final String extension = "csv";
+    private Integer lineNumber = 0;
+    private Integer attributesEnd = 0;
+    private List<String> lines = new ArrayList<String>();
+    private List<String> metadata = new ArrayList<String>();
+    private List<String> attributeMetadata = new ArrayList<String>();
+    private List<Map> attributes = new ArrayList<Map>();
+    private StringBuilder csvString = new StringBuilder();
+
+    private final String NEW_LINE = System.getProperty("line.separator");
+    private final String COMMENT = "# ";
+    private static final Logger logger = LoggerFactory.getLogger(GoGoDuck.class);
+
+    public CSVConverter() {}
+
+    @Override
+    public void init() {
+        this.ncksPath = GoGoDuckConfig.ncksPath;
+        this.ncpdqPath = GoGoDuckConfig.ncpdqPath;
+    }
+
+    @Override
+    public String getMimeType() { return mimeType; }
+
+    @Override
+    public String getExtension() { return extension; }
+
+    @Override
+    public Path convert(Path outputFile) throws GoGoDuckException {
+        List<String> ncpdqCommand = new ArrayList<>();
+        List<String> ncksCommand = new ArrayList<>();
+        String baseFilename = FilenameUtils.getBaseName(String.valueOf(outputFile));
+        try {
+            File tmpNcdf = File.createTempFile("tmp", ".nc");
+            // Unpack packed CSV values and store in temp csv
+            ncpdqCommand.add(ncpdqPath);
+            ncpdqCommand.add("-U");
+            ncpdqCommand.add("-O");
+            ncpdqCommand.add(outputFile.toString());
+            ncpdqCommand.add(tmpNcdf.getPath());
+            GoGoDuck.execute(ncpdqCommand);
+
+            // Output temp csv and store values
+            ncksCommand.add(ncksPath);
+            ncksCommand.add("--no_nm_prn");
+            ncksCommand.add("-a");
+            ncksCommand.add(tmpNcdf.getPath());
+
+            return executeNcks(ncksCommand, baseFilename, tmpNcdf.toPath());
+        } catch (Exception e) {
+            throw new GoGoDuckException(String.format("Could not convert output to csv: '%s'", e.getMessage()));
+        }
+    }
+
+    private Path executeNcks(List<String> commandList, String baseFilename, Path tempFile) throws IOException, InterruptedException {
+        logger.info(commandList.toString());
+
+        File errorTmpFile = File.createTempFile("error", ".txt");
+        ProcessBuilder processor = new ProcessBuilder(commandList);
+        processor.redirectError(errorTmpFile);
+
+        final Process process = processor.start();
+        InputStream is = process.getInputStream();
+        InputStreamReader isr = new InputStreamReader(is);
+        BufferedReader br = new BufferedReader(isr);
+        StringBuilder builder = new StringBuilder();
+
+        String line;
+        while ((line = br.readLine()) != null) {
+            builder.append(line + "\n");
```

Comment:
Probably a better way to obtain the newline separator for the system (instead of using `\n`).